isomerpages/isomercms-backend
# feat: create form to send email to all site collaborators

## Problem



Closes ISOM-1019.

## Solution



**Breaking Changes**




**Features**:

- Add support for a FormSG form that can automate the sending of emails to all collaborators of a site
    - Note: There is a technical limitation whereby Postman only can send to one recipient, so an alternative was done whereby the rest of the site admins are placed under the "CC" field.

## Tests



- Log in to the Staging CMS and go to the `jacob-staging` site
- Add yourself as a collaborator (provided that you are an Isomer Admin), remove the others if needed
- Fill in the staging form here: https://form.gov.sg/6640dc87dd10e94f6bd4ae40 (use "DNS records are incorrect (apex domain)" option)
- Fill in the DNS Records table as you wish (this will be shown verbatim inside the email)
- Submit the form and wait for the email to come into your inbox.
- Repeat for the other 2 options ("DNS records are incorrect (main domain)" and "Others"), verify that the emails are sent to your mailbox correctly.
    - Note: This will also create 3 separate emails to Isomer support, make sure to let Ops know that these are testing emails.

## Deploy Notes




**New environment variables**:

- `NOTIFY_SITE_COLLABORATORS_FORM_KEY ` : Form secret key for the notify site collaborators form
    - [x] added env var to 1PW + SSM script (`fetch_ssm_parameters.sh`)
